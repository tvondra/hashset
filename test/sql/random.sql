SELECT setseed(0.12345);

\set MAX_INT 2147483647

CREATE TABLE hashset_random_numbers AS
    SELECT
        (random()*:MAX_INT)::int AS i
    FROM generate_series(1,(random()*10000)::int)
;

SELECT
    md5(hashset_sorted)
FROM
(
    SELECT
        hashset_sorted(hashset(format('{%s}',string_agg(i::text,','))))
    FROM hashset_random_numbers
) q;

SELECT
    md5(input_sorted)
FROM
(
    SELECT
        format('{%s}',string_agg(i::text,',' ORDER BY i)) AS input_sorted
    FROM hashset_random_numbers
) q;
