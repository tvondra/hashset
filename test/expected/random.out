SELECT setseed(0.12345);
 setseed 
---------
 
(1 row)

\set MAX_INT 2147483647
CREATE TABLE hashset_random_numbers AS
    SELECT
        (random()*:MAX_INT)::int AS i
    FROM generate_series(1,(random()*10000)::int)
;
SELECT
    md5(hashset_sorted)
FROM
(
    SELECT
        hashset_sorted(hashset(format('{%s}',string_agg(i::text,','))))
    FROM hashset_random_numbers
) q;
               md5                
----------------------------------
 4ad6e4233861becbeb4a665376952a16
(1 row)

SELECT
    md5(input_sorted)
FROM
(
    SELECT
        format('{%s}',string_agg(i::text,',' ORDER BY i)) AS input_sorted
    FROM hashset_random_numbers
) q;
               md5                
----------------------------------
 4ad6e4233861becbeb4a665376952a16
(1 row)

